store.go

```diff
@@ -148,17 +151,19 @@ func (c *controller) getNetworksFromStore() ([]*network, error) {
 }
 
 func (n *network) getEndpointFromStore(eid string) (*endpoint, error) {
-	store := n.ctrlr.getStore(n.Scope())
-	if store == nil {
-		return nil, fmt.Errorf("could not find endpoint %s: datastore not found for scope %s", eid, n.Scope())
-	}
-
-	ep := &endpoint{id: eid, network: n}
-	err := store.GetObject(datastore.Key(ep.Key()...), ep)
-	if err != nil {
-		return nil, fmt.Errorf("could not find endpoint %s: %v", eid, err)
+	for _, store := range n.ctrlr.getStores() {
+		ep := &endpoint{id: eid, network: n}
+		err := store.GetObject(datastore.Key(ep.Key()...), ep)
+		// Continue searching in the next store if the key is not found in this store
+		if err != nil {
+			if err != datastore.ErrKeyNotFound {
+				log.Debugf("could not find endpoint %s: %v", eid, err)
+			}
+			continue
+		}
+		return ep, nil
 	}
-	return ep, nil
+	return nil, fmt.Errorf("could not find endpoint %s", eid)
```

Comment:
This code is now not correctly propagating the error if any in `GetObject`. If it doesn't find the object because of `ErrKeyNotFound` this is fine. But if the operation failed for some other reason we lose this information and will be hard troubleshooting such problems. Can we somehow capture the `GetObject` error if it is not `KeyNotFound`?